### PR TITLE
fix encoding of params for NFT encodings

### LINKED ISSUE #1518: ENS 721 token transfer function fail， build2.12 300
D3EDFF6E-79DD-4AFA-8C1B-CFE0A3568497

### PR DESCRIPTION
**Pull request by James-Sangalli**
Fixes #1518 & https://github.com/AlphaWallet/alpha-wallet-ios/issues/1438

### CODE REVIEW

**Comment by hboon** [comment]
@James-Sangalli there you go. I only went as far as stepping past the encoding. See if it makes sense and please test it :)

**Comment by James-Sangalli** [comment]
@hboon the encoding works but the contract address check is wrong (checking the recipient but not the contract address) this is bizarre but I will solve it here too.

**Comment by James-Sangalli** [comment]
@hboon fixed, see commit

File: AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift [hunk 1]
```diff
@@ -198,11 +198,11 @@ class TransactionConfigurator {
                 completion(.failure(AnyError(Web3Error(description: "malformed tx"))))
             }
 
-        case .ERC721Token:
+        case .ERC721Token(let token):
             do {
                 let function: Function
                 let parameters: [Any]
-                if transaction.to!.isLegacy721Contract {
```
Comment (James-Sangalli) [inline AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift]: @hboon this is the recipient address which is wrong

Comment (hboon) [reply AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift]: So bizzarre.

**Pull request merged**